package.json

```diff
@@ -3,14 +3,16 @@
   "version": "1.0.0-22",
   "description": "clojure.spec for Javascript",
   "main": "dist/js.spec.js",
+  "types": "types/index.d.ts",
   "scripts": {
     "build": "webpack --config webpack.config.js",
     "lint": "yarn run lint-src && yarn run lint-test",
     "lint-src": "eslint --config .eslintrc.json 'index.js' 'lib/**/*.js'",
     "lint-test": "eslint --config .eslintrc.test.json 'test/**/*.js'",
-    "prepare": "yarn run lint && yarn run build",
+    "prepare": "yarn run lint && yarn run test-ts && yarn run build",
     "test": "mocha --recursive --compilers js:babel-register test",
     "test:watch": "mocha --recursive --reporter=min --watch --compilers js:babel-register test",
+    "test-ts": "./node_modules/.bin/tsc --noImplicitAny --noEmit types/types-test.ts",
```

Comment:
I would add `test-ts` to both `test` and `test-watch` so that it is part of our testing hooks that the CI can call for each PR.